pilot/cmd/pilot-agent/main.go

```diff
@@ -208,6 +210,11 @@ var (
 			proxyConfig.ProxyAdminPort = int32(proxyAdminPort)
 			proxyConfig.Concurrency = int32(concurrency)
 
+			proxyConfig.Sds = &meshconfig.SDS{
```

Comment:
Please pass those directly, the meshconfig change should be reverted ( we are moving to use the mesh config defined part of operator, the change in mesh config uses different and redundant names/values and is not user editable)

Reply:
Besides - this will not be needed anyways, using detection and having a default value means we don't need it. 

If the injector adds the UDS path and the file exists - there is no need for additional agent flags or complexity.

Reply:
this isn't meshconfig but proxyconfig, code is simpler using arguments;
using env var also works, I don't have strong preference on this, new PR on pilot-agent will use env var as https://github.com/istio/istio/issues/15268